battles/forms.py

```diff
@@ -1,166 +1,162 @@
+from pokemons.helpers import is_pokemons_sum_valid
 from pokemons.models import Pokemon
+from users.models import User
 
 
 class CreateBattleForm(forms.ModelForm):
-    pokemon_1 = forms.CharField()
-    pokemon_2 = forms.CharField()
-    pokemon_3 = forms.CharField()
+    pokemon_1 = forms.ModelChoiceField(
+        queryset=Pokemon.objects.all(),
+        widget=autocomplete.ModelSelect2(
+            url='battles:pokemon_autocomplete',
+            attrs={
+                'data-placeholder': 'Autocomplete pokemon name...',
+                'data-minimum-input-length': 3,
+                'data-html': True,
+            },
+        )
+    )
+    pokemon_2 = forms.ModelChoiceField(
+        queryset=Pokemon.objects.all(),
+        widget=autocomplete.ModelSelect2(
+            url='battles:pokemon_autocomplete',
+            attrs={
+                'data-placeholder': 'Autocomplete pokemon name...',
+                'data-minimum-input-length': 3,
+                'data-html': True,
+            },
+        )
+    )
+    pokemon_3 = forms.ModelChoiceField(
+        queryset=Pokemon.objects.all(),
+        widget=autocomplete.ModelSelect2(
+            url='battles:pokemon_autocomplete',
+            attrs={
+                'data-placeholder': 'Autocomplete pokemon name...',
+                'data-minimum-input-length': 3,
+                'data-html': True,
+            },
+        )
+    )
+    order_1 = forms.ChoiceField(choices=POKEMON_ORDER_CHOICES, initial=0, required=True)
+    order_2 = forms.ChoiceField(choices=POKEMON_ORDER_CHOICES, initial=1, required=True)
+    order_3 = forms.ChoiceField(choices=POKEMON_ORDER_CHOICES, initial=2, required=True)
 
     class Meta:
         model = Battle
-        fields = ('trainer_opponent', 'trainer_creator', 'status')
-        widgets = {
-            'trainer_creator': forms.HiddenInput(),
-            'status': forms.HiddenInput(),
-        }
+        fields = ('trainer_opponent', )
 
     def __init__(self, *args, **kwargs):
-        initial = kwargs.get('initial', {})
-        initial_user_id = initial.get('user') and initial.get('user').id
-        initial['trainer_creator'] = initial.get('trainer_creator', initial_user_id)
-        initial['status'] = initial.get('status', 'ON_GOING')
-        kwargs['initial'] = initial
         super().__init__(*args, **kwargs)
+        users = User.objects.exclude(id=self.initial['trainer_creator'].id)
+        self.fields['trainer_opponent'].queryset = users
 
     def clean(self):
         cleaned_data = super().clean()
-        pokemons = [
-            self.cleaned_data['pokemon_1'].lower().strip(),
-            self.cleaned_data['pokemon_2'].lower().strip(),
-            self.cleaned_data['pokemon_3'].lower().strip()
+        rounds = [
+            int(self.cleaned_data['order_1']),
+            int(self.cleaned_data['order_2']),
+            int(self.cleaned_data['order_3']),
         ]
+        if len(Counter(rounds)) != 3:
```

Comment:
@lucabezerra I used `Counter` here because rounds list receives pokemons orders/rounds choices `(1-first, 2-second, 3-third)` and I wanted to guarantee that these three fields must be different, so if I take it off the `Counter`, the `length` would always be 3, regardless order choices fields content.

Reply:
Oh, makes sense! :)